project.php

```diff
@@ -248,6 +248,43 @@ function decide_blurbs()
     if ( $blurb )
         return array( $blurb, $blurb );
 
+    // Check whether the user is blocked by a daily page limit.
+    // Arguably, you'd expect this to be determined in can_user_get_pages_in_project(),
+    // but we also want to produce a warning (when the project is covered by a DPL)
+    // even when the user *isn't* blocked, which you can't really expect of that function.
+    //
+    $blocked_by_limit_message = "";
+    $page_limit_warning = "";
+    if ($round->has_a_daily_page_limit())
+    {
+        $user_dpl_count = $round->get_dpl_count_for_user($pguser);
+        if ($user_dpl_count >= $round->daily_page_limit)
+        {
+            // User has reached this round's DPL.
+            $blocked_by_limit_message = sprintf(
+                _("This project contributes to the '%s' limit of %d page-saves per day, and you have already reached that limit today."),
+                $round->id,
+                $round->daily_page_limit
+            );
+        }
+        else
+        {
+            // User hasn't reached this round's DPL,
+            // but they should be warned that there *is* a DPL for this round?
+            $msg = sprintf(
```

Comment:
I think we should only warn the user when they are at some percentage of the limit. Perhaps 90%?

We should also output this in our warning CSS class and probably with a `<p>` tag so it stands out from the surrounding text.

Reply:
90% sounds good to me.

Reply:
90%: Done.

'warning' class: Done. (And also for the "already reached the limit" message.)

`<p>` tag: I used `<span>` instead, because the "blurbs" mostly use `<br>` rather than `<p>`. (Not that I object to `<p>`, but it feels like that should be the subject of a separate refactoring.)

Reply:
LGTM!